### ISSUE TITLE
Substitute Callable and AsyncResult for their generic parameters automatically

### ISSUE THREAD

**Issue by zmitrok** [martypitt/swagger-springmvc#592]
Since this library is about Spring MVC integration, wouldn't it make sense to substitute Callable and AsyncResult models for their generic parameters by default?

**Comment by adrianbk** [comment]
@zmitrok sounds reasonable, any interest in submitting a PR for this?

**Comment by asm0dey** [comment]
Do you have any idea on how to implement substitution of Callable with real type?

**Comment by zmitrok** [comment]
Yes, take a look at genericModelSubstitutes method in the Fluent API of the Swagger MVC plugin

**Comment by asm0dey** [comment]
If I understand correctly that's the way you can transform types from your code. But I think it should be done somewhere inside springfox.
